### ISSUE TITLE
False alarm `TRIO900` when async context manager defined inside an async function

### ISSUE THREAD

**Issue by Zac-HD** [Zac-HD/flake8-trio#133]
```python
from contextlib import asynccontextmanager

async def this_is_not_an_async_generator():  # False alarm: TRIO900 on this line
    @asynccontextmanager
    async def cm():
        yield

    async with cm():
        pass
```
